### PR TITLE
Update to use YAML

### PR DESCRIPTION
**Pull request by froxcey**
YAML is a simpler file format for translators to work with, so I'm updating the translation library to read from YAML. As the time of writing, I haven't fully tested all features, and I still need to review and update documentation/guideline. All JsMap are already migrated to YAML, but they might have some errors. Changes are welcome, but please don't merge yet. I need to sleep now, and I will finalise this PR tomorrow.

Edit: It's now ready.

vibe

### CODE REVIEW

**Comment by froxcey** [comment]
Translation guideline got updated, memory leak fixed, key loaded by default, and all bugs should be gone now. I think this is ready for a Code Reviewed™!

File: .vscode/extensions.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "recommendations": ["esbenp.prettier-vscode"]
+  "recommendations": ["esbenp.prettier-vscode", "redhat.vscode-yaml"]
```
Comment (AZProductions) [inline .vscode/extensions.json:2]: Yes, this is awesome for new contributors

**Review by AZProductions — APPROVED** [review]
LGTM, yaml is definitely better and more understandable, thank you for making this pr 😊

**Comment by froxcey** [comment]
I see pull requests with in the old JsMap. I will edit them to use the new format.

**Pull request merged**